pkg/compactor/bucket_compactor.go

```diff
@@ -674,6 +674,23 @@ func IsCriticalError(err error) (bool, CriticalError) {
 	return ok, criticalErr
 }
 
+// IsPostingsOffsetTableSizeError returns true if the error indicates the postings offset
+// table size of the result block would exceed 4GB. This uses string matching because
+// Prometheus TSDB does not expose this as a structured error type.
+//
+// The error originates from vendor/github.com/prometheus/prometheus/tsdb/index/index.go (writeLengthAndHash).
```

Comment:
[Nit] Isn't there a risk of this assertion being fragile, i.e. the error origin could change in Prometheus?